libnest/src/package/mod.rs

```diff
@@ -59,6 +81,22 @@ impl CategoryName {
     }
 }
 
+impl std::ops::Deref for CategoryName {
+    type Target = String;
+
+    #[inline]
+    fn deref(&self) -> &String {
+        &self.0
+    }
+}
+
+impl std::convert::AsRef<Path> for CategoryName {
```

Comment:
Why did you implemented `AsRef<Path>` for something that logically doesn't represent a Path? (Same for `PackageName` and `RepositoryName`)